app/common/ts_util.c

```diff
@@ -0,0 +1,736 @@
+  if (!viewer) {
+    ierr = PetscViewerASCIIGetStdout(PetscObjectComm((PetscObject)ts),&viewer);CHKERRQ(ierr);
+  }
+  PetscValidHeaderSpecific(viewer,PETSC_VIEWER_CLASSID,2);
+  PetscCheckSameComm(ts,1,viewer,2);
+
+  ierr = PetscObjectTypeCompare((PetscObject)viewer,PETSCVIEWERASCII,&iascii);CHKERRQ(ierr);
+  ierr = PetscObjectTypeCompare((PetscObject)viewer,PETSCVIEWERSTRING,&isstring);CHKERRQ(ierr);
+  ierr = PetscObjectTypeCompare((PetscObject)viewer,PETSCVIEWERBINARY,&isbinary);CHKERRQ(ierr);
+  ierr = PetscObjectTypeCompare((PetscObject)viewer,PETSCVIEWERDRAW,&isdraw);CHKERRQ(ierr);
+
+  if (iascii) {
+  } else if (isstring) {
+  } else if (isbinary) {
+    PetscInt    classid = TS_FILE_CLASSID;
+    MPI_Comm    comm;
+    PetscMPIInt rank;
+    char        type[256];
+
+    ierr = PetscObjectGetComm((PetscObject)ts,&comm);CHKERRQ(ierr);
+    ierr = MPI_Comm_rank(comm,&rank);CHKERRQ(ierr);
+    if (!rank) {
+      ierr = PetscViewerBinaryWrite(viewer,&classid,1,PETSC_INT);CHKERRQ(ierr);
+      ierr = PetscStrncpy(type,((PetscObject)ts)->type_name,256);CHKERRQ(ierr);
+      ierr = PetscViewerBinaryWrite(viewer,type,256,PETSC_CHAR);CHKERRQ(ierr);
+
+      /* PETSC FIX START */
+      /* missing content */
+      ierr = PetscViewerBinaryWrite(viewer,&ts->steps,1,PETSC_INT);CHKERRQ(ierr);
+      ierr = PetscViewerBinaryWrite(viewer,&ts->ptime,1,PETSC_DOUBLE);CHKERRQ(ierr);
+      ierr = PetscViewerBinaryWrite(viewer,&ts->time_step,1,PETSC_DOUBLE);CHKERRQ(ierr);
+      ierr = PetscViewerBinaryWrite(viewer,&ts->ptime_prev,1,PETSC_DOUBLE);CHKERRQ(ierr);
+      ierr = PetscViewerBinaryWrite(viewer,&ts->ptime_prev_rollback,1,PETSC_DOUBLE);CHKERRQ(ierr);
+      ierr = PetscViewerBinaryWrite(viewer,&ts->solvetime,1,PETSC_DOUBLE);CHKERRQ(ierr);
+      /* PETSC FIX END */
+    }
+    if (ts->ops->view) {
+      ierr = (*ts->ops->view)(ts,viewer);CHKERRQ(ierr);
+    }
+    /* PETSC BUG */
+    /*if (ts->adapt) {ierr = TSAdaptView(ts->adapt,viewer);CHKERRQ(ierr);}*/
+    /* PETSC FIX START */
+    if (!ts->adapt) {
+      ierr = TSAdaptCreate(PetscObjectComm((PetscObject)ts),&ts->adapt);CHKERRQ(ierr);
+      ierr = TSAdaptSetType(ts->adapt,TSADAPTNONE);CHKERRQ(ierr);
+    }
+    ierr = TSAdaptView(ts->adapt,viewer);CHKERRQ(ierr);
+    
+    /* PETSC FIX END */
+    ierr = DMView(ts->dm,viewer);CHKERRQ(ierr);
+    ierr = VecView(ts->vec_sol,viewer);CHKERRQ(ierr);
+    ierr = DMGetDMTS(ts->dm,&sdm);CHKERRQ(ierr);
+    ierr = DMTSView(sdm,viewer);CHKERRQ(ierr);
+
+    ierr = _TSAdaptMembersView(ts->adapt,viewer);CHKERRQ(ierr);
+
+  } else if (isdraw) {
+  }
+  if (ts->snes && ts->usessnes)  {
+    ierr = PetscViewerASCIIPushTab(viewer);CHKERRQ(ierr);
+    ierr = SNESView(ts->snes,viewer);CHKERRQ(ierr);
+    ierr = PetscViewerASCIIPopTab(viewer);CHKERRQ(ierr);
+  }
+  /*ierr = DMGetDMTS(ts->dm,&sdm);CHKERRQ(ierr);*/
+  /*ierr = DMTSView(sdm,viewer);CHKERRQ(ierr);*/
+
+  ierr = PetscViewerASCIIPushTab(viewer);CHKERRQ(ierr);
```

Comment:
It seems these 3 lines are not doing anything useful?

Reply:
Yep, you're right.